generic_server.cc

```diff
@@ -372,6 +372,7 @@ future<> server::do_accepts(int which, bool keepalive, socket_address server_add
                         "too many in-flight connection attempts: {}, connection dropped",
                         _conns_cpu_concurrency_semaphore.waiters());
                 conn->shutdown().ignore_ready_future();
```

Comment:
should we await `conn->shutdown()`?

Reply:
also, it's unclear why the connection needs to be created and then shutdown, rather than not created at all.
I guess it signals overload to the client, but better document it in a comment at least.

Reply:
added comment, for await it's the other way around, no need to return future<> at all.